packages/mobx/src/types/observablearray.ts

```diff
@@ -237,7 +238,7 @@ export class ObservableArrayAdministration
         } else {
             const res = this.values_.slice(index, index + deleteCount)
             let oldItems = this.values_.slice(index + deleteCount)
-            this.values_.length = index + newItems.length - deleteCount
```

Comment:
Could you elaborate a bit on the rationale of this change and add a unit test with a splice larger then MAX_SPLICE_SIZE that also deletes items? Maybe this assignment should happen after the for loop, but not correcting for deleteCount sounds incorrect to me? Maybe this original assignment is faulty as well though. Looking at it, I think it should read:

`this.values_.length = this.values_.length + newItems.length - deleteCount;`. That `index` there looks very off :)

Reply:
You are correct! I didn't completely understand what was happening, now I do after another read-through